### DISCUSSION TITLE
Translation.json -> Two questions:  1. Comma on first string-position? 2. How to find the context

### DISCUSSION THREAD

**Discussion by n1smithy** [Ylianst/MeshCentral#6239] [Q&A] [answered]
Hi, MeshCentral-people!

I want to improve the German translation, but as German is a kind of high-precise and -complicated Language I need to know the context of some strings. Is there a chance to find the purposes of the single strings?

Aaaand what does a comma in the begining of a string mean?
`"de": ", zum Aktivieren hier klicken."`

How can I see if this is just a further part of a sentence or a phrase standing on its own?

I'm new in this so I need a little help. Thanks!

**Comment by si458** [comment]
if you look at the `"en"` value, you can see what the english version is,
you can then search the english phrase/sentence in any of handlebar files
so in this case above, its finding the english text below
`<div class="warningbox">Redirection port or KVM feature is disabled<span id="p11warninga">, click here to enable it.</span></div>`
so the comma at the beginning is needed as its seperates text from the original sentence

**Reply by n1smithy** [reply]
Thank you, that already helped at a lot of translations – even some still can be improved.

I'm on a Linux-Docker-machine. How can I use my own translation for testing? And some translations are still a blind shot because some contexts could'nt be resolved.

**Reply by n1smithy** [reply]
Oh, and one more question: are special keys like &nbsp; possible within the translated texts?

**Reply by si458** [reply, marked as answer]
https://ylianst.github.io/MeshCentral/how-to-contribute/#contribute-to-meshcentrals-multilingual-support
